scheduler/pkg/server/server_status.go

```diff
@@ -38,6 +38,14 @@ func (s *SchedulerServer) SubscribeModelStatus(req *pb.ModelSubscriptionRequest,
 	s.modelEventStream.mu.Unlock()
 
 	ctx := stream.Context()
+
+	go func() {
```

Comment:
it is still clear why this is required given that the same logic is done on line 24?

Reply:
You are right, only one of them should be kept.

This commit is base on v2.6 and rebases on v2 recently. In v2.6 sendCurrentModelStatuses is called after the steam added to modelEventStream.streams, that was fixed in your [commit](https://github.com/SeldonIO/seldon-core/commit/b05bcbaca09f297a878a5b91c246eb856bbdbb31) last month, I'll keep your new version.